wporg-gp-translation-events.php

```diff
@@ -428,5 +432,25 @@ public function pre_wp_unique_post_slug( $override_slug, string $slug, int $post
 		}
 		return $override_slug;
 	}
+
+	/**
+	 * Check if the current site is a translate.wordpress.org or a development TLD.
+	 *
+	 * @return bool
+	 */
+	private function is_dot_org_or_dev_tld(): bool {
+		// Allowed TLDs for development and testing.
+		$allowed_tlds = array( 'test', 'example', 'invalid', 'localhost' );
+		$site_url     = get_site_url();
+		$parsed_url   = wp_parse_url( $site_url );
+		$host         = $parsed_url['host'];
+
+		$domain_parts = explode( '.', $host );
+		$tld          = end( $domain_parts );
+		if ( 'translate.wordpress.org' === $host || in_array( $tld, $allowed_tlds, true ) ) {
+			return true;
+		}
+		return false;
+	}
```

Comment:
I use another local host format for this and it will be hard to include everyone. What if we did this?
```suggestion
	private function should_load_new_design(): bool {
		return 'translate.wordpress.org' === $host || (defined( 'TRANSLATION_EVENTS_NEW_DESIGN' ) && TRANSLATION_EVENTS_NEW_DESIGN );
	}
	
```